src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java

```diff
@@ -12,21 +16,68 @@ public MessageDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Message create(Message entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+
+            session.save(entity);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+                throw new RuntimeException(
+                        "Can't create message " + entity, e);
+            }
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return entity;
     }
 
     @Override
     public Message get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Message.class, id);
+        } catch (HibernateException e) {
+            throw new RuntimeException(
+                    "Can't get message with id = " + id + " from DB", e);
+        }
     }
 
     @Override
     public List<Message> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            Query<Message> allMessages = session.createQuery("from Message", Message.class);
```

Comment:
Ask mentor about naming. I think that here and in the other places must be like this:
```suggestion
            Query<Message> allMessagesQuery = session.createQuery("from Message", Message.class);
```